libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/GuidanceViewImageProvider.kt

```diff
@@ -83,16 +66,31 @@ class GuidanceViewImageProvider() {
 
     /**
      * The API allows you to cancel the rendering of guidance view.
-     * @param context Context
      */
-    fun cancelRender(context: Context) {
-        if (targets.isNotEmpty()) {
-            for(target in targets) {
-                Picasso.Builder(context).build().cancelRequest(target)
+    fun cancelRender() {
+        mainJobController.job.cancelChildren()
+    }
+
+    private suspend fun getBitmap(url: String): GuidanceViewImageResponse =
+        withContext(Dispatchers.IO) {
+            val req = Request.Builder().url(url).build()
+            try {
+                val res = okHttpClient.newCall(req).execute()
```

Comment:
our `Dispatchers.IO` has only 2 threads, so it might be issues when we block one of them with `call.execute`. 
we should use `call.enqueue` and `okHttp` will use one of his worker threads. 
to suspend current function until callback is fired use `suspendCoroutine`
the whole method will be

```
private suspend fun getBitmap(url: String): GuidanceViewImageResponse =
        withContext(Dispatchers.IO) {
            suspendCoroutine {
                val req = Request.Builder().url(url).build()
                okHttpClient.newCall(req).enqueue(object : Callback {
                    override fun onFailure(call: Call, e: IOException) {
                        resumeCoroutine(GuidanceViewImageResponse(error = e.message))
                    }

                    override fun onResponse(call: Call, response: Response) {
                        resumeCoroutine(
                            GuidanceViewImageResponse(
                                BitmapFactory.decodeStream(response.body()?.byteStream())
                            )
                        )
                    }

                    fun resumeCoroutine(result: GuidanceViewImageResponse) {
                        it.resumeWith(Result.success(result))
                    }
                })
            }
        }

    internal data class GuidanceViewImageResponse(
        val bitmap: Bitmap? = null,
        val error: String? = null
    )
```
I've also added default params to `GuidanceViewImageResponse`
please note, that in both `onResponse` and `onFailure` we use `resumeWith(Result.success(...))`

Reply:
@abhishek1508 ^^